File: client-app/src/app/models/Competence.ts
import { CompetenceProfiles } from "./CompetenceProfiles";

export interface Competence {
    id: string,
    name: string,
    description: string, 
    category: string,
    creationDate: Date | null,
    users: CompetenceProfiles[]
}

export class Competence implements Competence {
    constructor(init?: CompetenceFormValues) {
        Object.assign(this, init);
    }
}

export class CompetenceFormValues {
    id?: string = undefined;
    name: string = '';
    description: string = '';
    category: string = '';
    creationDate: Date | null = null;

    constructor(competence? : CompetenceFormValues) {
        if (competence) {
            this.id = competence.id;
            this.name = competence.name;
            this.description = competence.description;
            this.category = competence.category;
            this.creationDate = competence.creationDate;
        }
    }
}

File: Domain/Competence.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Competence
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public DateTime CreationDate { get; set; }
        public ICollection<UserCompetence> Users { get; set; } = new List<UserCompetence>();
        public ICollection<IssueCompetence> Issues { get; set; } = new List<IssueCompetence>();
    }
}